internal/pkg/template/template_functions.go

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package template
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	dashReplacement = "DASH"
+)
+
+// ReplaceDashes takes a CloudFormation logical ID, and
+// sanitizes it by removing "-" characters (not allowed)
+// and replacing them with "DASH" (allowed by CloudFormation but
+// not permitted in ecs-cli generated resource names).
+func ReplaceDashes(logicalID string) string {
+	return strings.ReplaceAll(logicalID, "-", dashReplacement)
+}
+
+// DashReplacedLogicalIDToOriginal takes a "sanitized" logical ID
+// and converts it back to its original form, with dashes.
+func DashReplacedLogicalIDToOriginal(safeLogicalID string) string {
+	return strings.ReplaceAll(safeLogicalID, dashReplacement, "-")
+}
+
+var nonAlphaNum = regexp.MustCompile("[^a-zA-Z0-9]+")
+
+// StorageLogicalIDSafe strips non-alphanumeric characters from an input string.
+func StorageLogicalIDSafe(s string) string {
```

Comment:
Should we name this just:
```suggestion
func LogicalIDSafe(s string) string {
```